src/backend/distributed/commands/foreign_server.c

```diff
@@ -0,0 +1,126 @@
+/*-------------------------------------------------------------------------
+ *
+ * foreign_server.c
+ *    Commands for FOREIGN SERVER statements.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "catalog/pg_foreign_server.h"
+#include "distributed/commands/utility_hook.h"
+#include "distributed/commands.h"
+#include "distributed/deparser.h"
+#include "distributed/listutils.h"
+#include "distributed/metadata/distobject.h"
+#include "distributed/metadata_sync.h"
+#include "distributed/worker_transaction.h"
+#include "foreign/foreign.h"
+#include "nodes/primnodes.h"
+
+
+List *
+PreprocessCreateForeignServerStmt(Node *node, const char *queryString,
+								  ProcessUtilityContext processUtilityContext)
+{
```

Comment:
It's important to guard these functions with an if (ShouldPropagate())

Reply:
Missed that, added, thank you!